### PR TITLE
MC-7404: VMWare template details - update docs

### PR DESCRIPTION
**Pull request by SeventhOfNine**
#### Issue
When importing a template, there are extra optional parameters that can be set when selecting VMWare as the hypervisor.
Add the following new parameters:
- NIC Adapter Type
- Root Disk Controller
- Keyboard

#### Solution
Update the documentation to reflect the new parameters.
Add:
- import template
- update template
- delete template

### CODE REVIEW

File: source/includes/compute/_templates.md [hunk 1]
```diff
@@ -16,39 +16,59 @@ curl -X GET \
...
-`availableInZones`<br/>*array[UUID]* | List of all [zone ids](#zones) that the template is available in
+`osTypeId`<br/>*UUID* | Id of the OS type
+`osTypeName`<br/>*string* | The OS type of the template (e.g. Ubuntu, CentOS...)
+`zoneId`<br/>*UUID* |The [zone id](#cloudstack-zones) that the template is available in. Empty if available in multiple zones
+`zoneName`<br/>*string* |The [zone id](#cloudstack-zones) that the template is available in. Empty if available in multiple zones
+`availableInZones`<br/>*array[UUID]* | List of all [zone ids](#cloudstack-zones) that the template is available in
+`hypervisor`<br>*string* | The name of the hypervisor
+`format`<br>*string* | The template format for the chosen hypervisor
+`nicAdapter`<br>*string* | `VMWare only` The network interface controller name. Empty if not selected.<br>*Values*: e1000, pcnet32, vmxnet2 or vmxnet3.
+`rootDiskController`<br>*string* | `VMWare only` The root disk controller name. Empty if not selected.<br>*Values*: ide, scsi for CloudStack version up to 4.4.<br>*Values*: ide, scsi, osdefault, pvscsi, lsilogic, lsisas1068, buslogic for CloudStack version 4.5 and up.
```
Comment (pdube) [inline source/includes/compute/_templates.md]: VMWare isn't supported in cca so I would remove all fields relative to it

**Review by pdube — CHANGES_REQUESTED** [review]

File: source/includes/compute/_templates.md [hunk 2]
```diff
@@ -16,39 +16,59 @@ curl -X GET \
...
-    "osType": "Other (64-bit)",
+    "zoneId":"ea901007-056b-4c50-bb3a-2dd635fce2ab",
+    "zoneName": "zone1",
+    "osTypeId":"d38dc1d9-dd34-4fdd-8f51-a973f8fe5f3b",
+    "osTypeName": "Other (64-bit)",
     "availableInZones": [
        "ea901007-056b-4c50-bb3a-2dd635fce2ab"
-    ]
+    ],
+    "hypervisor": "VMware",
```
Comment (simongodard) [inline source/includes/compute/_templates.md]: We should not mention VMware-specific fields for cloud.ca since that hypervisor is not supported.

**Review by simongodard — CHANGES_REQUESTED** [review]
Same comment as @pdube about VMware fields

File: source/includes/compute/_templates.md (hunk 2 above)
Comment (SeventhOfNine) [reply source/includes/compute/_templates.md]: Good point I will remove them.

**Review by simongodard — APPROVED** [review]

**Review by pdube — APPROVED** [review]

File: source/includes/compute/_templates.md (hunk 1 above)
Comment (SeventhOfNine) [reply source/includes/compute/_templates.md]: Removed.

File: source/includes/compute/_templates.md (hunk 2 above)
Comment (SeventhOfNine) [reply source/includes/compute/_templates.md]: Removed.

**Pull request merged**